pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -0,0 +1,381 @@
+// to corresponding filter config that is used by the envoy RBAC filter to enforce access control to
+// the service co-located with envoy.
+// Currently the config is only generated for sidecar node on inbound HTTP listener. The generation
+// is controlled by RbacConfig (a singleton custom resource defined in istio-system namespace). User
+// could disable this by either deleting the RbacConfig or set the RbacConfig.mode to OFF.
+// Note: This is still working in progress and by default no RbacConfig is created in the deployment
+// of Istio which means this plugin doesn't generate any RBAC config by default.
+package authz
+
+import (
+	"sort"
+	"strings"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	rbacconfig "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/rbac/v2"
+	http_conn "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	policyproto "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2alpha"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+	"istio.io/istio/pilot/pkg/serviceregistry/kube"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	// RbacFilterName is the name of the RBAC filter in envoy.
+	RbacFilterName = "envoy.filters.http.rbac"
+
+	// RbacConfigName is the name of the RbacConfig custom resource that controls the RBAC behavior.
+	RbacConfigName = "rbac-config"
+
+	sourceIP        = "source.ip"
+	sourceService   = "source.service"
+	destinationIP   = "destination.ip"
+	destinationPort = "destination.port"
+	userHeader      = ":user"
+	methodHeader    = ":method"
+	pathHeader      = ":path"
+	//TODO(yangminzhu): Update the key after the final header name of service is decided.
+	serviceHeader       = ":service"
+	requestHeaderPrefix = "request.header["
+	requestHeaderSuffix = "]"
+)
+
+// Plugin implements Istio RBAC authz
+type Plugin struct{}
+
+// NewPlugin returns an instance of the authz plugin
+func NewPlugin() plugin.Plugin {
+	return Plugin{}
+}
+
+// OnOutboundListener is called whenever a new outbound listener is added to the LDS output for a given service
+// Can be used to add additional filters on the outbound path
+func (Plugin) OnOutboundListener(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
+	return nil
+}
+
+// OnInboundListener is called whenever a new listener is added to the LDS output for a given service
+// Can be used to add additional filters (e.g., mixer filter) or add more stuff to the HTTP connection manager
+// on the inbound path
+func (Plugin) OnInboundListener(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
+	// Only supports sidecar proxy of HTTP listener for now.
+	if in.Node.Type != model.Sidecar || in.ListenerType != plugin.ListenerTypeHTTP {
+		return nil
+	}
```

Comment:
Why?

Reply:
Currently Istio RBAC only applies to HTTP listeners on sidecars, no ingress or TCP filters are supported for now.